src/components/map/Layers/Countries.tsx

```diff
@@ -0,0 +1,113 @@
+import { useState, useEffect, useContext } from "react"
+import { AppContext } from "context";
+import ReactDOMServer from "react-dom/server";
+import { EstimateGradePrevalence } from "types";
+import mapboxgl from "mapbox-gl";
+import CountryPopup from 'components/map/Popups/CountryPopup'
+
+function SetCountryEstimates(map: mapboxgl.Map, estimateGradePrevalences: EstimateGradePrevalence[]) {
+    estimateGradePrevalences.forEach((country: EstimateGradePrevalence) => {
+        if (country && country.testsAdministered && country.alpha3Code) {
+            map.setFeatureState(
+                {
+                    source: "Countries",
+                    sourceLayer: "Countries",
+                    id: country.alpha3Code,
+                },
+                {
+                    hasData: true,
+                    testsAdministered: country.testsAdministered,
+                    geographicalName: country.geographicalName,
+                    numberOfStudies: country.numberOfStudies,
+                    localEstimate: country.localEstimate,
+                    nationalEstimate: country.nationalEstimate,
+                    regionalEstimate: country.regionalEstimate,
+                    sublocalEstimate: country.sublocalEstimate,
+                }
+            );
+        }
+    });
+}
+
+function UpdateCountryEstimates(map: mapboxgl.Map, estimateGradePrevalences: EstimateGradePrevalence[]) {
+    var onlyIso3: (string | null)[] = estimateGradePrevalences.map((c) => { return c.alpha3Code })
+
+    if (map.getLayer("Countries") && map.isSourceLoaded('Countries')) {
+        map?.setFilter('Countries',
+            ["in",
+                ["get", 'CODE'],
+                ["literal", onlyIso3]
+            ]);
+    }
+    else { // If countries has not been loaded, wait until it's source has finished.
+        map.on('sourcedata', function (e: any) {
+            if (e.sourceId == 'Countries' && map.isSourceLoaded('Countries')) {
+                map?.setFilter('Countries',
+                    ["in",
+                        ["get", 'CODE'],
+                        ["literal", onlyIso3]
+                    ]);
+            }
+        });
+    }
+}
+
+const Countries = (map: mapboxgl.Map | undefined, estimateGradePrevalences: EstimateGradePrevalence[]) => {
+
+    const [state] = useContext(AppContext);
+    const [popup, setPopup] = useState<mapboxgl.Popup | undefined>(undefined);
+
+    useEffect(() => {
```

Comment:
and can we add some quick comments around what the purpose of each useEffect is? just so that we don't get confused about the data flow in the future